### PR TITLE
Asynchronous Oracle

### PR DESCRIPTION
**Pull request by shargon**
Require: https://github.com/neo-project/neo/pull/1738 https://github.com/neo-project/neo-devpack-dotnet/pull/348

### CODE REVIEW

File: src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma warning disable CS0626
+
+namespace Neo.SmartContract.Framework.Services.Neo
+{
+    [Contract("0x3c05b488bf4cf699d0631bf80190896ebbf38c3b")]
+    public class Oracle
+    {
+        public static extern byte[][] GetOracleNodes();
+        public static extern void Request(string url, string filter, string callback, object userData, long gasForRepsonse);
```
Comment (EdgeDLT) [inline src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:9]: `gasForRepsonse` should be `gasForResponse`?

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:9]: Yes, thanks!

File: src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma warning disable CS0626
+
+namespace Neo.SmartContract.Framework.Services.Neo
+{
+    [Contract("0x3c05b488bf4cf699d0631bf80190896ebbf38c3b")]
+    public class Oracle
+    {
+        public static extern string Name();
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:8]: Use property?

File: src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        public static extern byte[][] GetOracleNodes();
```
Comment (Tommo-L) [inline src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:9]: Add `SetOracleNodes` ?

Comment (Tommo-L) [reply src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:9]: And we also need some way to define the `Callback` interface.

Comment (erikzhang) [reply src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:9]: `SetOracleNodes` is never used by normal users. I think we don't need to provide it.

**Review by Tommo-L — APPROVED** [review]

**Comment by Tommo-L** [comment]
Ping? 
 I need this for testing oracle plugin.

**Review by erikzhang — APPROVED** [review]

File: src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs (hunk 2 above)
Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/Oracle.cs:8]: I will do it in other PR for allow @Tommo-L to use these changes

**Pull request merged**